### PR TITLE
Content warning fixups

### PR DESCRIPTION
**Pull request by matc-pub**
- Require consent of logged in existing users.
- Prevent link clicks in blurred areas.
- Enable tippy again.
- Prevent stacking of countdowns for Back button.
- Fix auto expand blocking for images.

### CODE REVIEW

File: src/server/handlers/catch-all-handler.tsx [hunk 1]
```diff
@@ -145,7 +145,7 @@ export default async (req: Request, res: Response) => {
       errorPageData,
       showAdultConsentModal:
         !!site?.site_view.site.content_warning &&
-        !(site.my_user || req.cookies[adultConsentCookieKey]),
```
Comment (SleeplessOne1917) [inline src/server/handlers/catch-all-handler.tsx:148]: Logged in users should not be shown the consent modal.

File: src/shared/components/app/app.tsx [hunk 2]
```diff
@@ -23,6 +24,14 @@ export default class App extends Component<any, any> {
...
   private readonly rootRef = createRef<HTMLDivElement>();
 
+  componentDidMount() {
+    setupTippy(this.rootRef);
+  }
+
+  componentWillUnmount() {
+    destroyTippy();
+  }
+
```
Comment (SleeplessOne1917) [inline src/shared/components/app/app.tsx:34]: Doh! I can't believe I removed those. Good catch.

File: src/shared/components/post/post-listing.tsx [hunk 3]
```diff
@@ -103,7 +103,7 @@ export class PostListing extends Component<PostListingProps, PostListingState> {
   private readonly isoData: IsoDataOptionalSite = setIsoData(this.context);
   state: PostListingState = {
     showEdit: false,
-    imageExpanded: !!this.isoData.site_res?.site_view.site.content_warning,
```
Comment (SleeplessOne1917) [inline src/shared/components/post/post-listing.tsx:106]: Images are supposed to auto expand when content warning is present.

**Review by SleeplessOne1917 — CHANGES_REQUESTED** [review]
The code is fine, but some of your changes go against requirements.

File: src/shared/components/post/post-listing.tsx (hunk 3 above)
Comment (matc-pub) [reply src/shared/components/post/post-listing.tsx:106]: Not the opposite? Collapsed while the modal is showing. Expanded when auto expand is enabled and the image is not nsfw or the user settings disable the blur for nsfw.

File: src/server/handlers/catch-all-handler.tsx (hunk 1 above)
Comment (SleeplessOne1917) [reply src/server/handlers/catch-all-handler.tsx:148]: I was going by [lemmynsfw's changes](https://github.com/LemmyNet/lemmy-ui/compare/main...lemmynsfw:lemmy-ui:main) for this one. I doubt instances that will need the consent form will shift to using it after the instance has been established long enough to have a userbase.

File: src/shared/components/post/post-listing.tsx (hunk 3 above)
Comment (SleeplessOne1917) [reply src/shared/components/post/post-listing.tsx:106]: I'm using lemmynsfw's behavior as a reference, but TBH this does seem like something that should be a separate site setting.

File: src/server/handlers/catch-all-handler.tsx (hunk 1 above)
Comment (dessalines) [reply src/server/handlers/catch-all-handler.tsx:148]: I pry agree that logged-in users don't need to be shown the popup. If they're already part of the instance they know what its about.

File: src/shared/components/post/post-listing.tsx (hunk 3 above)
Comment (dessalines) [reply src/shared/components/post/post-listing.tsx:106]: related to this: https://github.com/LemmyNet/lemmy/issues/4643

I'd prefer we just let the new `local_site.default_post_listing_mode` handle this, and add a card view.

**Review by SleeplessOne1917 — APPROVED** [review]

**Pull request merged**